src/ucp/tag/rndv.c

```diff
@@ -16,27 +16,31 @@
 #include <ucs/datastruct/queue.h>
 
 static int ucp_rndv_is_recv_pipeline_needed(ucp_request_t *rndv_req,
-                                            uintptr_t remote_address,
+                                            const ucp_rndv_rts_hdr_t *rndv_rts_hdr,
                                             ucs_memory_type_t mem_type)
 {
     ucp_md_index_t md_index;
     uct_md_attr_t *md_attr;
+    ucp_md_map_t md_map;
 
     /* no bw lanes */
     if (!ucp_ep_config(rndv_req->send.ep)->key.rma_bw_md_map) {
         return 0;
     }
 
     /* disqualify recv side pipeline if
-     * a bw lane exist, which can register the mem_type AND
-     * lane can't do HOST RMA if remote is HOST mem type.
+     * a mem_type bw lane exist AND
+     * lane can do RMA on remote mem_type
      */
+    md_map = UCS_BIT(mem_type);
+    if (rndv_rts_hdr->address) {
+        md_map |= UCS_BIT(ucp_rkey_packed_mem_type(rndv_rts_hdr + 1));
```

Comment:
ucp_rkey_packed_mem_type returns memory type
why is it legal to add to to memory domains bitmap?

Reply:
my bad. misleading variable name. it should be "mem_type_map".
can I fix the variable name and merge all commits in the next push?

Reply:
> can I fix the variable name and merge all commits in the next push?

let's squash only after review is finished